### PR TITLE
Add Java versions 9~14 to E2E tests

### PR DESCRIPTION
**Pull request by kezhenxu94**

### CODE REVIEW

File: .github/workflows/e2e.jdk-versions.yaml [hunk 1]
```diff
@@ -47,6 +47,8 @@ jobs:
           git submodule sync --recursive
           git -c protocol.version=2 submodule update --init --force --recursive --depth=1
       - name: Build Docker Image
+        env:
+          JAVA_VERSION: ${{ matrix.jdk }}
```
Comment (wu-sheng) [inline .github/workflows/e2e.jdk-versions.yaml:51]: So, agent and backend tests will run in the same JDK versions, right?

**Comment by wu-sheng** [comment]
@hanahmily I think you will like this :)

File: .github/workflows/e2e.jdk-versions.yaml (hunk 1 above)
Comment (kezhenxu94) [reply .github/workflows/e2e.jdk-versions.yaml:51]: > So, agent and backend tests will run in the same JDK versions, right?

Yes, the agent and backend should be independent in JDK versions, so just to make it simple

**Comment by wu-sheng** [comment]
A lot of Tests fail. Please confirm.

**Comment by wu-sheng** [comment]
@hanahmily This PR changes `bash` to `sh`, do you have any concern about this?

**Comment by kezhenxu94** [comment]
> @hanahmily This PR changes `bash` to `sh`, do you have any concern about this?

@hanahmily the real concern is that I change the base docker image to a slim one

File: pom.xml [hunk 2]
```diff
@@ -255,6 +255,12 @@
             <version>${lombok.version}</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>javax.annotation</groupId>
+            <artifactId>javax.annotation-api</artifactId>
+            <version>1.3.2</version>
```
Comment (hanahmily) [inline pom.xml:261]: Add a property?

Comment (wu-sheng) [reply pom.xml:261]: Add a core level dependency for?

**Comment by hanahmily** [comment]
@kezhenxu94 Greate job! except for some nits, these changes look good to me.

File: pom.xml (hunk 2 above)
Comment (hanahmily) [reply pom.xml:261]: jdk11 has removed it from rt.jar, that's he add this dependency here.

Comment (kezhenxu94) [reply pom.xml]: > Add a core level dependency for?

Just to make it compile in JDK 9+, the scope is `provided` so nothing changed actually

**Pull request merged**